src/engine/Actor.ts

```diff
@@ -521,6 +519,12 @@ export class ActorImpl extends Class implements IActionable, IEvented, IPointerE
          child._initialize(engine);
       }
    }
+   // Initialize default options, that depends on Environment defaults
+   private _initDefaults() {
+      console.log(Engine);
+      // this.anchor = Engine.defaults.actorAnchor;
+      this.anchor = Vector.Half;
```

Comment:
When referencing the `Vector.Constants` we should also `.clone()`

`this.anchor = Vector.Half.clone()`

Reply:
Since these aren't value types in javascript we can run into trouble with subtle bugs sharing vector references, it's just good practice to `.clone()` vectors when you can.

Reply:
So, there is the point, where all the fuss has stared!
Idea, which i wanted to introduce was a little rethink of @kamranayub original one, presented in #705 . And this part is just some code to represent the issue. 
Mine idea is commented one: `// this.anchor = Engine.defaults.actorAnchor;`

The point is to have an `Engine.defaults` property, as @kamranayub . And restrict this property for changing for everyone, except `Engine`. 
By default every created `Actor` has their `Actor.anchor` set to `Engine.defaults.actorAnchor` and scope for this property is `Global`. It means, that we can only change `Engine.default.actorAnchor` to value we need, and every `Actor` with `Global` scope will change this value. And if we do not need some `Actor` to change his value, we could just simply change his scope to `Local`. Like, 'Actor.setLocalAnchor(Vector)` or smth like that. Isn't it meaningless?

Reply:
@LoserAntbear I see where you're coming from, but I don't think it's necessary to block changing the default except when on engine. 

If a game developer wants all anchors to be changed `Actor.defaults.anchor` could be set at start of the game, then all actors created after that would get the new default. I think it would be unexpected to set the `Actor.defaults.anchor` then have every actor in the game simultaneously update, I feel like that would violate the "principle of least surprise".

If an Actor needs a different anchor, the developers can have their individual anchors updated already with the `.anchor` property. Currently it is possible to do this:

```typescript
// game start
let engine = new ex.Engine({width: 600, height: 400});
Actor.defaults.anchor = new ex.Vector(.75, .75);
game.start(() => {

...

// defaults anchor to new ex.Vector(.75, .75) at constructor time
let actor = new ex.Actor({x: 10, y: 20, width: 50, height: 50, color: ex.Color.Red.clone()})

// set to something else if needed
actor.anchor = new ex.Vector(0, 0);

game.add(actor)

});

```